shipengine_sdk/models/address/__init__.py

```diff
@@ -14,19 +22,60 @@ class Address:
     state_province: str
     postal_code: str
     country_code: str
-    is_residential: bool = False
+    is_residential: Optional[bool] = None
     name: str = ""
     phone: str = ""
     company: str = ""
 
+    def __post_init__(self):
+        is_street_valid(self.street)
+        is_city_valid(self.city_locality)
+        is_state_valid(self.state_province)
+        is_postal_code_valid(self.postal_code)
+        is_country_code_valid(self.country_code)
+
 
-@dataclass_json(letter_case=LetterCase.CAMEL)
-@dataclass(frozen=True)
 class AddressValidateResult:
     is_valid: Optional[bool]
     request_id: str
     normalized_address: Optional[Address]
-    messages: List
-    # info: List
-    # warnings: List
-    # errors: List
+    info: Optional[List]
+    warnings: Optional[List]
+    errors: Optional[List]
+
+    def __init__(
+        self,
+        is_valid: Optional[bool],
+        request_id: str,
+        normalized_address: Optional[Address],
+        messages: List,
+        info: Optional[List] = None,
+        warnings: Optional[List] = None,
+        errors: Optional[List] = None,
+    ):
+        self.is_valid = is_valid
+        self.request_id = request_id
+        self.normalized_address = normalized_address
+        self.info = list() if info is None else info
+        self.warnings = list() if warnings is None else warnings
+        self.errors = list() if errors is None else errors
+        self.__extract_messages(messages)
+
+    def __extract_messages(self, messages):
+        for message in messages:
+            if "type" in message:
+                if message["type"] == "error":
+                    del message["type"]
```

Comment:
I think this function maybe probably be improved with some higher level constructs like map  / chunk / filter /  partition etc. :-)